logger 2-phase build fails in Docker 1.3

I tried with Ubuntu Docker 1.3.0 and Mac boot2docker 1.3.0. `make -C logger/ build` doesn't work any longer. (This is on the `docker build` host and unrelated to #2166.)

``` console
Unable to find image 'logger-build' locally
2014/10/17 17:26:50 Error: image library/logger-build not found
docker build -t logger-build .
Sending build context to Docker daemon 38.91 kB
Sending build context to Docker daemon 
Step 0 : FROM golang:1.3
golang:1.3: The image you are pulling has been verified
50215b109eda: Pull complete 
53f380325ee9: Pull complete 
ad97f49020c4: Pull complete 
e4fb9076d684: Pull complete 
abc0add39a58: Pull complete 
0543a3778018: Pull complete 
ac69e0684bde: Pull complete 
95201034386b: Pull complete 
8b0171d2e8a7: Pull complete 
601ea7bdf704: Pull complete 
6982015c482b: Pull complete 
b235f28e14c9: Pull complete 
511136ea3c5a: Already exists 
Status: Downloaded newer image for golang:1.3
 ---> b235f28e14c9
Step 1 : WORKDIR /go/src/github.com/deis/deis/logger
 ---> Running in 7a75d4e5ddf9
 ---> 771ea116bb4c
Removing intermediate container 7a75d4e5ddf9
Step 2 : CMD /go/bin/logger
 ---> Running in 7cc73aad0720
 ---> 2a55e637e576
Removing intermediate container 7cc73aad0720
Step 3 : ADD . /go/src/github.com/deis/deis/logger
 ---> 7ef8156f9ae8
Removing intermediate container ed50618fe8cb
Step 4 : RUN CGO_ENABLED=0 go get -a -ldflags '-s' github.com/deis/deis/logger
 ---> Running in 164148db6290
 ---> b5b10a93d96e
Removing intermediate container 164148db6290
Successfully built b5b10a93d96e
docker cp Pulling repository logger-build:/go/bin/logger image/bin/

Usage: docker cp CONTAINER:PATH HOSTPATH

Copy files/folders from the PATH to the HOSTPATH

make[1]: *** [build] Error 2
make: *** [build] Error 2
```

--- a/logger/Makefile
+++ b/logger/Makefile
@@ -7,11 +7,10 @@ DEV_IMAGE = $(DEV_REGISTRY)/$(IMAGE)
 
 build: check-docker
 	docker build -t $(BUILD_IMAGE) .
-	@$(eval CID := $(shell docker run -d $(BUILD_IMAGE)))
-	docker cp $(CID):/go/bin/logger image/bin/
+	docker cp `docker run -d $(BUILD_IMAGE)`:/go/bin/logger image/bin/
 	docker build -t $(IMAGE) image
-	-docker kill $(CID)
 	rm -rf image/bin/logger
+	-docker rm -f `docker ps | grep logger-build | awk '{print $$1}'`
 
 clean: check-docker check-registry
 	docker rmi $(IMAGE)

